Fix nodeinfo query on split-domain instances

Drakulix opened the pull request:
Fixes https://github.com/cheeaun/phanpy/issues/1073

Drakulix commented:
> This seems to assume that URL structure is always http://nodeinfo.diaspora.software/ns/schema/[VERSION]. I'm not sure if we want to handle cases if the URL is http://nodeinfo.diaspora.software/ns/schema/[VERSION]/something#maybe?

The [spec](https://github.com/jhass/nodeinfo/blob/main/PROTOCOL.md) doesn't seem to allow this for all known variants, so it might be something that could be added later? But sure I can fix that.

> .reverse() is not needed if the values are already flipped from within the .sort() function. E.g. .sort((a, b) => -compareVersions(a.version, b.version));

👍 

> .find((x) => satisfies(x.version, '<=2.1'))?.href; seems to force it to always be <=2.1? This wouldn't be future-compatible if there's a newer version later?

That is on purpose. To quote the spec:
> A client should follow the link matching the highest schema version it supports.

We can't be sure any new variants still carry the fields we are using and servers are encouraged to support multiple versions. E.g. gotosocial already publishes links for both `2.0` and `2.1`.

cheeaun commented:
> > This seems to assume that URL structure is always [http://nodeinfo.diaspora.software/ns/schema/[VERSION]](http://nodeinfo.diaspora.software/ns/schema/%5BVERSION%5D). I'm not sure if we want to handle cases if the URL is [http://nodeinfo.diaspora.software/ns/schema/[VERSION]/something#maybe](http://nodeinfo.diaspora.software/ns/schema/%5BVERSION%5D/something#maybe)?
> 
> The [spec](https://github.com/jhass/nodeinfo/blob/main/PROTOCOL.md) doesn't seem to allow this for all currently known variants, so it might be something that could be added later? But sure I can fix that.

Yeah, maybe later 🤞👍

> > .find((x) => satisfies(x.version, '<=2.1'))?.href; seems to force it to always be <=2.1? This wouldn't be future-compatible if there's a newer version later?
> 
> That is on purpose. To quote the spec:
> 
> > A client should follow the link matching the highest schema version it supports.
> 
> We can't be sure any new variants still carry the fields we are using and servers are encouraged to support multiple versions. E.g. gotosocial already publishes links for both `2.0` and `2.1`.

I mean if one day there's v2.2 or v3.0, it should magically follow the highest version right, instead of hardcoding the max version? Unless a major version might be a breaking change 🙈.

Drakulix commented:
> I mean if one day there's v2.2 or v3.0, it should magically follow the highest version right, instead of hardcoding the max version? Unless a major version might be a breaking change 🙈.

I am not sure tbh. The spec doesn't say anything about this version number following semver or something else. So imo any future version could contain a breaking change, though I guess it is safe to assume, that removing/renaming a field would only happen on a major bump, so I changed the check to "<=2".

(Also fixed the `sort`+`reverse`.)
